BuildScript/BuildTargets/Testing.targets

```diff
@@ -201,80 +212,55 @@ $(_dockerImageName) $(_nunitCallArgumentString)
       <Output TaskParameter="Value" PropertyName="_nunit_failures" />
     </MSBuild.ExtensionPack.Xml.XmlFile>
 
-    <MSBuild.ExtensionPack.Xml.XmlFile
-        TaskAction="ReadAttribute"
-        File="$(_testResultFile)"
-        XPath="/test-results/@errors">
-      <Output TaskParameter="Value" PropertyName="_nunit_errors" />
-    </MSBuild.ExtensionPack.Xml.XmlFile>
-
     <MSBuild.ExtensionPack.Framework.TextString TaskAction="Replace" OldString="$(_testTime)" OldValue="." NewValue="">
       <Output TaskParameter="NewString" PropertyName="_testTime" />
     </MSBuild.ExtensionPack.Framework.TextString>
 
     <Message Text="##teamcity[buildStatisticValue key='$(_testName)' value='$(_testTime)']"
              Condition="'$(TEAMCITY_VERSION)' != ''"/>
 
-    <ItemGroup Condition="$(_nunit_failures) &gt; 0 or $(_nunit_errors) &gt; 0">
+    <ItemGroup Condition="$(_nunit_failures) &gt; 0">
       <NumberOfTestFailuresPerProject Include="$(_nunit_failures)"/>
-      <NumberOfTestErrorsPerProject Include="$(_nunit_errors)"/>
       <FailedProjects Include="$(_testAssemblyFullPath)">
         <Platform>$(_platform)</Platform>
         <DatabaseSystem>$(_databaseSystem)</DatabaseSystem>
         <Browser>$(_browser)</Browser>
         <Failures>$(_nunit_failures)</Failures>
-        <Errors>$(_nunit_errors)</Errors>
       </FailedProjects>
     </ItemGroup>
   </Target>
 
-  <!-- We need to configure Nunit differently to run it on a System with or without .Net 3.5 installed. Can be removed after the switch to Nunit 3-->
-  <Target Name="PrepareNunitDirectory" Outputs="%(_nunitDestinationFolderPath.Identity)">
+  <Target Name="PrepareNunitDirectory" Outputs="$(_nunitRunnerFolderPath)">
     <Error Text="The property 'TempDirectory' is not set." Condition="'$(TempDirectory)' == ''" />
     <Error Text="The property 'NUnitToolPath' is not set." Condition="'$(NUnitToolPath)' == ''" />
 
     <PropertyGroup>
-      <_dockerTempFolderName>$(TempDirectory)SharedSourceBuildNunitCopy\</_dockerTempFolderName>
-      <NunitFolderPathWithDefaultConfiguration>$(_dockerTempFolderName)NunitDefaultConfiguration\</NunitFolderPathWithDefaultConfiguration>
-      <NunitFolderPathWithoutDotNet35Dependency>$(_dockerTempFolderName)NunitWithoutDotNet35Dependency\</NunitFolderPathWithoutDotNet35Dependency>
+      <_nunitRunnerFolderPath>$(TempDirectory)SharedSourceBuildNunitCopy\</_nunitRunnerFolderPath>
     </PropertyGroup>
 
-    <ItemGroup>
-      <_nunitDestinationFolderPath Include="$(NunitFolderPathWithDefaultConfiguration)">
-        <SupportedRuntimeConfiguration></SupportedRuntimeConfiguration>
-      </_nunitDestinationFolderPath>
-
-      <_nunitDestinationFolderPath Include="$(NunitFolderPathWithoutDotNet35Dependency)">
-        <SupportedRuntimeConfiguration>&lt;supportedRuntime version=&quot;v4.0.30319&quot; /&gt;</SupportedRuntimeConfiguration>
-      </_nunitDestinationFolderPath>
-    </ItemGroup>
+    <WriteLinesToFile
```

Comment:
What's that one doing? it's just creating a file with the contents
> *.dll

?

Reply:
`nunit.engine.dll` in `NUnitToolPath` looks for `.addins` file next to it (meaning in the same directory) to include NUnit extensions (like the V2 driver) when running tests. What I do here is create a custom `.addins` file that includes all `.dll` files inside the same directory, place it inside `NUnitToolPath` then I copy `NUnitToolPath` and `NUnitExtensionsPaths` to the same temporary directory, meaning NUnit itself and all NUnit extensions are together in one place.

NUnit then scans all `.dll` files if they are extensions and includes them in the test run if they are.